### PR TITLE
feat: add dataset operations to sdkserver

### PR DESCRIPTION
**Pull request by g-linville**
This adds five new operations to the SDKServer to run the tools that we are using as the API for datasets. The SDKServer is basically just acting as a middleman between the client using the SDK, and the dataset API (the five tools in https://github.com/gptscript-ai/datasets/pull/1).

### CODE REVIEW

File: pkg/gptscript/gptscript.go [hunk 1]
```diff
@@ -80,6 +84,9 @@ func Complete(opts ...Options) Options {
 	if len(result.CredentialContexts) == 0 {
 		result.CredentialContexts = []string{credentials.DefaultCredentialContext}
 	}
+	if result.DatasetToolRepo == "" {
+		result.DatasetToolRepo = defaultDatasetToolRepo
+	}
```
Comment (thedadams) [inline pkg/gptscript/gptscript.go:89]: I don't think this is needed given that you have a default set.

I do believe it is needed here, though, when running the SDK server in embedded mode: https://github.com/gptscript-ai/gptscript/blob/45d444f810600d9584951ec3aba98da9b5e02db5/pkg/sdkserver/server.go#L154

Comment (g-linville) [reply pkg/gptscript/gptscript.go]: Thanks. This is fixed

File: pkg/sdkserver/datasets.go [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+		writeError(logger, w, http.StatusInternalServerError, fmt.Errorf("failed to run program: %w", err))
+		return
+	}
+
+	writeResponse(logger, w, map[string]any{"stdout": result})
+}
+
+type createDatasetArgs struct {
+	Name        string `json:"dataset_name"`
+	Description string `json:"dataset_description"`
```
Comment (thedadams) [inline pkg/sdkserver/datasets.go:85]: nit: you have camelCase and snake_case in your `json` tags. Can this be standardized?

**Review by thedadams — APPROVED** [review]

File: pkg/sdkserver/datasets.go (hunk 2 above)
Comment (g-linville) [reply pkg/sdkserver/datasets.go:85]: fixed

Comment (thedadams) [reply pkg/sdkserver/datasets.go:85]: This went the opposite way I expected. Is there reason why you are doing snake_case? I think most of our stuff uses camelCase.

Comment (g-linville) [reply pkg/sdkserver/datasets.go:85]: Lol. I honestly have no idea. I'll go change it all to camelCase.

Comment (g-linville) [reply pkg/sdkserver/datasets.go]: Fixed. Sorry for the delay. Got stuck working on a test for another repo

**Review by thedadams — APPROVED** [review]

**Review by njhale — APPROVED** [review]

**Pull request merged**